edge-modules/ModuleRestarter/src/Program.cs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace ModuleRestarter
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices;
+    using Microsoft.Azure.Devices.Edge.ModuleUtil;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    class Program
+    {
+        static readonly ILogger Logger = ModuleUtil.CreateLogger("ModuleRestarter");
+
+        public static int Main() => MainAsync().Result;
+
+        static async Task<int> MainAsync()
+        {
+            Logger.LogInformation($"Starting module restarter with the following settings:\r\n{Settings.Current}");
+
+            if (Settings.Current.DesiredModulesToRestart.Count == 0)
+            {
+                Logger.LogInformation("No modules names found in input. Stopping.");
+                return 0;

```

Comment:
If the module is stopped when no modules were set, but is deployed to be running. edgeAgent will start it.

Reply:
Restart policy is "on-failure"

Reply:
However, someone could deploy the module differently from long-haul settings, which could lead to unexpected behavior.